### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by baev**
### Misc

* Add global eslint config
* Add global prettier
* Husky to run prettier precommit hooks
* Fix code style according to meet new source code requirements

### Breaking

* Remove few deprecations (No more interfaces with I prefix)
    `AllureConfig` was removed
    `IAllureConfig` was renamed to `AllureConfig`
    `AllureWriter` was renamed to `FileSystemAllureWriter`
    `IAllureWriter` was renamed to `AllureWriter`

* All API methods now has return type specified

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

**Review by sseliverstov — APPROVED** [review]

File: packages/allure-jest/test/default-runner/simple.test.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// eslint-disable-next-line no-undef
```
Comment (just-boris) [inline packages/allure-jest/test/default-runner/simple.test.ts:1]: Should be fixed with `/*eslint-env jest*/`?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    "one-var": ["off", "never"],
+    "padding-line-between-statements": [
+      "off",
+      {
+        blankLine: "always",
+        prev: "*",
+        next: "return",
+      },
+    ],
+    "prefer-arrow/prefer-arrow-functions": "error",
```
Comment (just-boris) [inline .eslintrc.js:232]: Not sure about this one: `const demo = () => {}` is 21 character, but `function demo() {}` is 18. Not sure why someone would prefer a longer form.

There is a built-in "[prefer arrow callback](https://eslint.org/docs/rules/prefer-arrow-callback)" rule which makes sense, but using arrow for all is too much imo.

**Review by just-boris — COMMENTED** [review]
minor suggestions

File: packages/allure-jest/test/default-runner/simple.test.ts (hunk 1 above)
Comment (baev) [reply packages/allure-jest/test/default-runner/simple.test.ts]: done

File: .eslintrc.js (hunk 2 above)
Comment (baev) [reply .eslintrc.js:232]: it is alignment with our internal code style, so I like to keep it

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -26,9 +26,18 @@ jobs:
           npm run prepare
         env:
           CI: true
-      - name: npm build, test
+      - name: npm build
         run: |
           npm run build
+        env:
```
Comment (mxschmitt) [inline .github/workflows/build.yml:32]: FYI, `CI` is always set to `true`: https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

**Pull request merged**